content/tutorials/Super-lightweight-realtime-subscriptions-with-SSE-and-Ably.textile

```diff
@@ -0,0 +1,223 @@
+---
+title: Super-lightweight realtime subscriptions with SSE and Ably
+section: tutorials
+index: 109
+languages:
+  - javascript
+  - python
+  - php
+---
+
+h2(#what-is-sse). What is SSE?
+
+Server-sent events is a standard describing how servers can initiate data transmission towards clients once an initial client connection has been established. They are commonly used to send message updates or continuous data streams to a browser client and designed to enhance native, cross&#45;browser streaming through a JavaScript API called EventSource, through which a client requests a particular URL in order to receive an event stream. Basically, a server&#45;sent event is when a web page automatically gets updates from a server.
+
+h2(#setting-up-the-ui-and-publisher). Setting up the UI and publisher
+
+*Setting up the UI*
+
+Let us create a simple HTML file that implements SSE using Ably. Create a file called @example.html@ and add:
+
+```[html]
+
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+        <style>
+          ul {
+              width: 60%;
+              margin-top: 10px;
+              margin: 0 auto;
+              font-family: courier, courier new;
+              background-color: #333;
+              height: 300px;
+              overflow: scroll;
+              color: orange;
+          }
+          h1{
+            text-align: center;
+          }
+          h3{
+            text-align: center;
+          }
+    </style>
+  </head>
+  <body>
+    <h1>SSE example</h1>
+
+    <h3>Events:</h3>
+    <ul id="events"></ul>
+  </body>
+</html
+```
+
+What's happening in the code above? We create a basic HTML structure which consists of an @ul@ element. The children of this element will be dynamically populated.
```

Comment:
> What's happening in the code above? We

In the code above, we create a basic HTML page which consists of an unordered list @ul@ element.

> The children of this element will be dynamically populated.

The children of this element will be dynamically populated with the real-time event and data updates received from the publisher.